File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        long long temp = x, reversed = 0;
        if(x < 0)
        {
            return false;
        }
        else if (x > 0 && x < 10)
        {
            return true;
        }
        else
        {
            while(temp)
            {
                reversed = reversed * 10 + temp % 10;
                temp /= 10;
            }
            if (reversed == x)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
        
        
    }
};

File: 0009-palindrome-number/0009-palindrome-number.cs
public class Solution {
    public bool IsPalindrome(int x) {
        int reversedNum = 0;
        bool result;
        int temp = x;
        if(x < 0)
        {
            result = false;
        }
        else
        {
            while(temp != 0)
            {
                reversedNum = reversedNum * 10 + temp % 10;
                temp /= 10; 
            }
            if(x == reversedNum)
                result = true;
            else
            result = false;
        }

        return result;
        
    }
}